tools/ci_build/github/azure-pipelines/binary-size-checks-pipeline.yml

```diff
@@ -24,15 +29,18 @@ jobs:
   pool: Linux-CPU-2019
 
   strategy:
+    # Notes:
+    # - BuildConfigFile path is relative to repository root.
+    # - BinarySizeThresholdInBytes is optional - if unspecified it is not checked.
     matrix:
+      # required configurations
       AndroidMinimalBaseline:
-        ConfigId: minimal-baseline
-        BuildConfigFile: "/onnxruntime_src/tools/ci_build/github/linux/ort_minimal/build_check_binsize_config/android_minimal_baseline.config"
+        BuildConfigFile: "tools/ci_build/github/linux/ort_minimal/build_check_binsize_config/android_minimal_baseline.config"
         BinarySizeThresholdInBytes: 1306224
-      AndroidMinimalBasicWithSomeOps:
-        ConfigId: android-minimal-basic-with-some-ops
-        BuildConfigFile: "/onnxruntime_src/tools/ci_build/github/linux/ort_minimal/build_check_binsize_config/android_minimal_basic_with_some_ops.config"
-        BinarySizeThresholdInBytes: 2658304
+      # additional configurations
+      ${{ if or(eq(parameters.BuildAllConfigurations, true), and(in(variables['Build.Reason'], 'IndividualCI', 'BatchedCI'), eq(variables['Build.SourceBranch'], 'refs/heads/main'))) }}:
```

Comment:
For your information, not all variables are available for use in templates.  You can use https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml to look up what variables are available. The vars here are good.